sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java

```diff
@@ -396,6 +412,31 @@ public Builder withTrustMaterial(Set<X509Certificate> certificates, TrustAnchorT
             return withTrustMaterial(trustStore, trustOptions);
         }
 
+        public Builder withPlaceHolderIdentityMaterial() {
+            return withPlaceHolderMaterial(
+                    "javax.net.ssl.keyStore",
+                    "javax.net.ssl.keyStorePassword",
+                    "javax.net.ssl.keyStoreType",
+                    this::withIdentityMaterial
+            );
+        }
+
+        private Builder withPlaceHolderMaterial(String keyStorePathSystemProperty, String keyStorePasswordProperty, String keyStoreTypeProperty, TriConsumer<Path, char[], String> keyStorePropertyConsumer) {
+            Path keystore = Optional.ofNullable(System.getProperty(keyStorePathSystemProperty))
+                    .map(Paths::get)
+                    .orElse(null);
+
+            char[] keystorePassword = Optional.ofNullable(System.getProperty(keyStorePasswordProperty))
+                    .map(String::toCharArray)
+                    .orElse(null);
+
+            String keystoreType = Optional.ofNullable(System.getProperty(keyStoreTypeProperty))
+                    .orElseGet(KeyStore::getDefaultType);
+
```

Comment:
for completeness `javax.net.ssl.keyStoreProvider` should be also read/used here

Reply:
Ah, very sharp. I added it! Thank you